src/BuildScriptGenerator/DotNetCore/GlobalJsonSdkResolver.cs

```diff
@@ -0,0 +1,371 @@
+            {
+                return availableSdk.Major == versionToResolve.Major
+                && availableSdk.Minor == versionToResolve.Minor;
+            });
+
+            var sameFeatureSdks = availableSdks.Where(availableSdk => availableSdk.Feature == versionToResolve.Feature);
+            if (sameFeatureSdks.Any())
+            {
+                return GetLatestPatch(sameFeatureSdks, versionToResolve);
+            }
+            else
+            {
+                var higherFeatureSdks = availableSdks
+                    .Where(availableSdk => availableSdk.Feature > versionToResolve.Feature);
+                if (higherFeatureSdks.Any())
+                {
+                    var nextHigherFeatureSdk = higherFeatureSdks.OrderBy(sdk => sdk).First();
+                    var nextHigherFeatureSdks = higherFeatureSdks.Where(sdk => sdk.Feature == nextHigherFeatureSdk.Feature);
+                    return GetLatestPatch(nextHigherFeatureSdks, nextHigherFeatureSdk);
+                }
+            }
+
+            return null;
+        }
+
+        private string GetMinor(IEnumerable<SdkVersionInfo> availableSdks, SdkVersionInfo versionToResolve)
+        {
+            /*
+                Uses the latest patch level for the specified major, minor, and feature band.
+                If not found, rolls forward to the next higher feature band within the same major/minor version
+                and uses the latest patch level for that feature band.
+                If not found, rolls forward to the next higher minor and feature band within the same major and
+                uses the latest patch level for that feature band.
+                If not found, fails.
+            */
+            var result = GetFeature(availableSdks, versionToResolve);
+            if (!string.IsNullOrEmpty(result))
+            {
+                return result;
+            }
+
+            // If not found, rolls forward to the next higher minor and feature band within the same major and
+            // uses the latest patch level for that feature band.
+            availableSdks = availableSdks.Where(availableSdk =>
+            {
+                return availableSdk.Major == versionToResolve.Major;
+            });
+
+            if (availableSdks.Any())
+            {
+                var higherMinorSdks = availableSdks.Where(sdk => sdk.Minor > versionToResolve.Minor);
+                if (higherMinorSdks.Any())
+                {
+                    var nextHighestMinorSdk = higherMinorSdks.OrderBy(sdk => sdk).First();
+                    var nextHighestMinorSdks = higherMinorSdks.Where(sdk => sdk.Minor == nextHighestMinorSdk.Minor);
+                    return GetFeature(nextHighestMinorSdks, nextHighestMinorSdk);
+                }
+            }
+
+            return null;
+        }
+
+        private string GetMajor(IEnumerable<SdkVersionInfo> availableSdks, SdkVersionInfo versionToResolve)
```

Comment:
repeating my question here … 
as per this logic it looks like 
if user is locally using some non-preview version of 2.1.4 (for example)
we will try to get the latest patch for the same, if not we move forward with next higher feature band within 2.1 and so on. so in that case we are good accepting point 2 that you mentioned above.

can you confirm?

Reply:
Just to make it clear, this PR does not handle the scenario of the points (`1.` and `2.`) that I made above. I just wanted to bring up some points for consideration in a later PR (if you look at this PR it is not integrated with rest of Oryx yet)

> if user is locally using some non-preview version of 2.1.4 (for example)
we will try to get the latest patch for the same, if not we move forward with next higher feature band within 2.1 and so on. so in that case we are good accepting point 2 that you mentioned above.

I did not understand your scenario clearly but if a user did not provide a global.json file, we can decide on either doing `minor` or `latestMinor` based on a runtime version.

Reply:
ah ok

> Just to make it clear, this PR does not handle the scenario of the points (1. and 2.) that I made above. I just wanted to bring up some points for consideration in a later PR (if you look at this PR it is not integrated with rest of Oryx yet)
> 
> if user is locally using some non-preview version of 2.1.4 (for example)
> we will try to get the latest patch for the same, if not we move forward with next higher feature band within 2.1 and so on. so in that case we are good accepting point 2 that you mentioned above.
> 
> I did not understand your scenario clearly but if a user did not provide a global.json file, we can decide on either doing minor or latestMinor based on a runtime version.

im fine with current test coverage then